recurly/__init__.py

```diff
@@ -1267,25 +1267,54 @@ def convert_trial_moto(self):
 
     def convert_trial(self, three_d_secure_action_result_token_id = None):
         """Convert trial to paid subscription"""
-        account = recurly.Account()
-        billing_info = recurly.BillingInfo(
-            three_d_secure_action_result_token_id = three_d_secure_action_result_token_id
-        )
-        account.billing_info = billing_info
-
         url = urljoin(self._url, '/convert_trial')
+        
         if not three_d_secure_action_result_token_id == None:
             elem = ElementTreeBuilder.Element(self.nodename)
-            elem.append(Resource.element_for_value('account', account))
+            three_d_s_token = Resource.element_for_value("three_d_secure_action_result_token_id", three_d_secure_action_result_token_id)
+            billing_info = Resource.element_for_value("billing_info", None)
+            account = Resource.element_for_value("account", None)
+            
+            billing_info.append(three_d_s_token)
+            account.append(billing_info)
+            elem.append(account)
             body = ElementTree.tostring(elem, encoding='UTF-8')
-        
-            response = self.http_request(url, 'PUT', body, { 'Content-Type':
-                'application/xml; charset=utf-8' })
+        else:
+            body = None
+
+        response = self.http_request(url, 'PUT', body, { 'Content-Type':
+            'application/xml; charset=utf-8' })
+        print(f'Response: {response}')    
+
+        if response.status not in (200, 201, 204):
+            self.raise_http_error(response)
+
+        self.update_from_element(ElementTree.fromstring(response.read()))
+
+
+    # def convert_trial(self, three_d_secure_action_result_token_id = None):
+    #     """Convert trial to paid subscription"""
+    #     account = recurly.Account()
+    #     billing_info = recurly.BillingInfo(
+    #         three_d_secure_action_result_token_id = three_d_secure_action_result_token_id
```

Comment:
I think you should skip using Resources entirely and just build the xml directly. Example:

```python
request = ElementTreeBuilder.Element('subscription')
token_el = ElementTreeBuilder.SubElement(request, 'three_d_secure_action_result_token_id')
token_el.text = 'mytoken'
body = ElementTree.tostring(elem, encoding='UTF-8')
# => <subscription><three_d_secure_action_result_token_id>mytoken</three_d_secure_action_result_token_id></subscription>
```

You should be able to use the same technique to build the other request.